src/tokens/selectors.ts

```diff
@@ -56,6 +63,17 @@ export const tokensByIdSelector = createSelector(
       }
     }
     return tokenBalances
+  },
+  {
+    memoizeOptions: {
+      equalityCheck: (previousValue, currentValue) => {
+        if (isNetworkIdList(previousValue) && isNetworkIdList(currentValue)) {
+          return _.isEqual(previousValue, currentValue)
+        }
+        return previousValue === currentValue
```

Comment:
I opted for this approach since it seemed like the "right" solution imo -- rather than serializing/deserializing the array to work with the default reference equality checker, defining our own custom checker seemed like the cleaner solution.

Reply:
True, yes the custom equality check looks right, thanks 👍 
Maybe a micro optim would be to tag the networkIdList with a [symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) so we don't have to iterate over it. But likely doesn't matter. Just me overthinking this :D